logstreams/src/test/java/io/zeebe/logstreams/log/LogStreamBatchWriterTest.java

```diff
@@ -414,4 +414,16 @@ public void shouldNotFailToWriteBatchWithoutEvents() {
     // then
     assertThat(pos).isEqualTo(0);
   }
+
+  @Test
+  public void shouldFailToWriteOnClosedLogStream() {
+    // given
+    logStreamRule.getLogStream().close();
+
+    // when
+    final long pos = writer.event().key(1).value(EVENT_VALUE_1).done().tryWrite();
+
+    // then
+    assertThat(pos).isEqualTo(-1);
+  }
```

Comment:
Do we already have a test for creating reader or writer on closed log stream?

Reply:
We have one in LogStreamTest.